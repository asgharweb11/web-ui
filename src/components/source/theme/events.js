import theme from "./theme"

export const Color = (item , param) => {
    switch(item){
        case "ligth" :
            return theme.ligth.text.grey;
        case "dark" :
            return theme.dark.text.grey;
    }
}