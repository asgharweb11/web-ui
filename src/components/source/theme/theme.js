export default {
    ligth : {
        main : "white",
        content : "repeating-linear-gradient(0deg , #f2efff , #f3f3ff)",
        primary : "#2196f3",
        secondary : "#ff80ab",
        error : "#f44336",
        warning : "#ff9800",
        info : "#2196f3",
        success : "#4caf50",
        text : {
            grey : "#313131",
            grey1 : "#656565",
            grey2 : "#909090",
            grey3 : "#808080",
            grey4 : "#9a9a9a",
            grey5 : "#c7c7c7",
            color : "#ffffff",
        },
        glass : {
            grey : "#000000a3",
            grey2 : "#00000066",
            grey3 : "#0000003b",
            grey4 : "#0000001c",
            grey5 : "#0000000d",
        },
        hover : {
            primary : "#246fab",
            secondary : "#ec5286",
            error : "#dc1709",
            warning : "#e08703",
            info : "#0d7ad2",
            success : "#237b26",
        }
    },  
    dark : {
        main : "#212121",
        content : "repeating-linear-gradient(0deg ,#333333 ,#333333)",
        primary : "#4dabf5",
        secondary : "#ff99bb",
        error : "#e57373",
        warning : "#ffb74d",
        info : "#64b5f6",
        success : "#81c784",
        text : {
            grey : "#ffffff",
            grey1 : "#ececec",
            grey2 : "#c7c7c7",
            grey3 : "#212121",
            grey4 : "#6b6b6b",
            grey5 : "#464646",
            color : "2b2b2b"
        },
        glass : {
            grey : "#ffffffb3",
            grey2 : "#ffffff87",
            grey3 : "#ffffff59",
            grey4 : "#ffffff2b",
            grey5 : "#ffffff0f",
        },
        hover : {
            primary : "#90ceff",
            secondary : "#ffb5ce",
            error : "#fdaaaa",
            warning : "#ffd393",
            info : "#a8d8ff",
            success : "#b0efb2",
        }

    },
    fontSize : {
        h1 : "40px",
        xl : "25px",
        lg : "20px",
        md : "16px",
        sm : "13px",
        xs : "10px",
    },
    px : {
        p1 : "0px",
        p2 : "1px",
        p3 : "5px",
        p4 : "10px",
        p5 : "15px",
        p6 : "20px",
    },
    media : {
        mobileS: '320px',
        mobileM: '375px',
        mobileL: '425px',
        tablet: '768px',
        tabletL : "991px",
        laptop: '1024px',
        laptopM: '1200px',
        laptopL: '1440px',
        desktop: '2560px'
    }

}