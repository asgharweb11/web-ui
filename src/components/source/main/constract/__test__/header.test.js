import React from "react"
import renderer from "react-test-renderer"
// -------------- Context
import Context from "../../../../../context.js"
// -------------- Styled
import {ThemeProvider} from "styled-components"
import theme from "../../../theme/theme"
// -------------- Conponent
import Header from "../header"


test('component : constract/header.js', () => {
    const container = renderer.create(<Context>
        <ThemeProvider theme={theme}>
            <Header />
        </ThemeProvider>
    </Context>)

    let tree = container.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.children[0].children[0].children[1].children[0]).toEqual("Secondary")
})
