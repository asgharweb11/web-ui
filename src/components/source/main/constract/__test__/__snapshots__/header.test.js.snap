// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component : constract/header.js 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="sc-bdfBwQ hEOWYR header"
    >
      <h1>
        Delivery
        <span>
          Center
        </span>
      </h1>
      <button
        className="btn btn-outline-secondary"
        disabled={false}
        type="button"
      >
        Secondary
      </button>
    </div>
  </div>
</div>
`;
